### PR TITLE
Play audio

### PR DESCRIPTION
**Pull request by WilliamHarmonMS**
@joshlory  or @Hamms PTAL

we're playing pressurePlate and Piston sound effects. I think that's it so far? Still working on music.

Fixes https://github.com/code-dot-org/craft/issues/85 and https://github.com/code-dot-org/craft/issues/140.

### CODE REVIEW

File: src/js/game/LevelMVC/LevelPlane.js [hunk 1]
```diff
@@ -330,14 +330,29 @@ module.exports = class LevelPlane {
...
+    switch (sound) {
+      case "On":
+        this.levelModel.controller.audioPlayer.play("pistonOut");
+        break;
+      case "Off":
+        this.levelModel.controller.audioPlayer.play("pistonIn");
+        break;
+      case "":
+        //do nothing
+        break;
```
Comment (joshlory) [inline src/js/game/LevelMVC/LevelPlane.js]: Tiny nit: you can remove this case.

File: src/js/game/LevelMVC/LevelPlane.js [hunk 2]
```diff
@@ -378,20 +393,24 @@ module.exports = class LevelPlane {
...
+    let returnValue = "";
 
     if (block.blockType.startsWith("piston") && !block.blockType.startsWith("pistonArm")) {
       block.isPowered = this.powerCheck(position, true);
       if (block.isPowered) {
         this.activatePiston(position);
+        returnValue = "On";
       } else if (!block.isPowered) {
         this.deactivatePiston(position);
+        returnValue = "Off";
```
Comment (joshlory) [inline src/js/game/LevelMVC/LevelPlane.js]: Can you change these "On" -> "Out" and "Off" -> "In" so you don't need the `playPistonSound` switch statement above?

**Comment by joshlory** [comment]
Hmm, I can't seem to get the Piston to make a sound?

**Comment by WilliamHarmonMS** [comment]
@joshlory 😞 

You're totally right, Pistons weren't being caught in some cases because I wasn't saving the string properly and it was getting set back to null before the sound call was made. That's fixed now. We should be good!

**Comment by WilliamHarmonMS** [comment]
@joshlory PTAL fixed a merge conflict and this should be good to go ❤️

**Comment by joshlory** [comment]
Hmm, placing a block next to the pressure plate seems to re-trigger the piston sound.

**Comment by WilliamHarmonMS** [comment]
Good catch @joshlory 
I was trying to be clever and check for piston state changes, but I did it too high up and was playing the sound if pistons were powered / unpowered, regardless of what their -former- state was. This was in an effort to circumvent the playSound call being needed at the actual -activation / deactivation- calls, because if the sound gets called more than once (because more than 1 piston changes states as a result of the action) the sound got real bad real fast.

This is now fixed, but it either was going to involve passing return values up two or three functions, or just having a class-wide accessible flag that can be set in the activation/deactivation functions, but then only referenced and reset once after all the pistons are checked (this is the solution I went with).

**Comment by WilliamHarmonMS** [comment]
Yeah they ended up being easily overlap-able, but this DOES prevent multiple version of the same sound from triggering at the same time (ie. if you power 4 pistons, only 1 piston sound is played, not 4x of the same audio)

**Pull request merged**